core/predicate_check.go

```diff
@@ -18,50 +18,90 @@ import (
 
 var ErrMissingPredicateContext = errors.New("missing predicate context")
 
-// CheckPredicates verifies the predicates of [tx] and returns the result. Returning an error invalidates the block.
-func CheckPredicates(rules params.Rules, predicateContext *precompileconfig.PredicateContext, tx *types.Transaction) (predicate.PrecompileResults, error) {
-	// Check that the transaction can cover its IntrinsicGas (including the gas required by the predicate) before
-	// verifying the predicate.
-	intrinsicGas, err := IntrinsicGas(tx.Data(), tx.AccessList(), tx.To() == nil, rules)
+// CheckBlockPredicates verifies the predicates of a block of transactions and
+// returns the results.
+//
+// Returning an error invalidates the block.
+func CheckBlockPredicates(
+	rules params.Rules,
+	predicateContext *precompileconfig.PredicateContext,
+	txs []*types.Transaction,
+) (predicate.BlockResults, error) {
+	var results predicate.BlockResults
+	// TODO: Calculate tx predicates concurrently.
```

Comment:
Because the checks here are stateless, each predicate can be trivially verified concurrently.

I didn't make that change in this PR, but will in a follow-up.